src/token/api/src/state.rs

```diff
@@ -1,61 +1,72 @@
-use crate::ledger::Ledger;
-use crate::types::{Allowances, AuctionInfo, Cycles, Metadata, StatsData, Timestamp};
-use candid::{CandidType, Deserialize, Principal};
+use std::collections::HashMap;
+
+use candid::{CandidType, Deserialize, Int, Principal};
+use ic_helpers::ledger::{AccountIdentifier, Subaccount};
 use ic_helpers::tokens::Tokens128;
 use ic_storage::stable::Versioned;
 use ic_storage::IcStorage;
-use std::collections::HashMap;
+
+use crate::account::{Account, SUB_ACCOUNT_ZERO};
+use crate::ledger::Ledger;
+use crate::types::{AuctionInfo, Claims, Cycles, Metadata, StatsData, Timestamp, TxError, Value};
 
 #[derive(Debug, Default, CandidType, Deserialize, IcStorage)]
 pub struct CanisterState {
     pub bidding_state: BiddingState,
     pub balances: Balances,
     pub auction_history: AuctionHistory,
     pub stats: StatsData,
-    pub allowances: Allowances,
     pub ledger: Ledger,
+    pub claims: Claims,
 }
 
 impl CanisterState {
+    pub fn icrc1_metadata(&self) -> HashMap<String, Value> {
+        let mut metadata = HashMap::new();
+        metadata.insert(
+            "icrc1:symbol".to_string(),
+            Value::Text(self.stats.symbol.clone()),
+        );
+        metadata.insert(
+            "icrc1:name".to_string(),
+            Value::Text(self.stats.name.clone()),
+        );
+        metadata.insert(
+            "icrc1:decimals".to_string(),
+            Value::Int(Int::from(self.stats.decimals)),
+        );
+        metadata
```

Comment:
would be nice to include owner/ feeto information here, principal as string

Reply:
OK , on it